From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 2 Oct 2012 17:25:46 +0200
Subject: [10/12] [SCSI] virtio-scsi: initialize scatterlist structure

commit 2e9c9dfde00a6466441e93033cf2c37f720bdacf upstream.

The sg struct is used without being initialized, which breaks
when CONFIG_DEBUG_SG is enabled.

Signed-off-by: Richard W.M. Jones <rjones@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
---
 drivers/scsi/virtio_scsi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-3.2.46/drivers/scsi/virtio_scsi.c
===================================================================
--- linux-3.2.46.orig/drivers/scsi/virtio_scsi.c	2013-07-26 19:27:52.000000000 +0000
+++ linux-3.2.46/drivers/scsi/virtio_scsi.c	2013-07-26 19:27:54.000000000 +0000
@@ -219,7 +219,7 @@
 	struct scatterlist sg;
 	unsigned long flags;
 
-	sg_set_buf(&sg, &event_node->event, sizeof(struct virtio_scsi_event));
+	sg_init_one(&sg, &event_node->event, sizeof(struct virtio_scsi_event));
 
 	spin_lock_irqsave(&vscsi->event_vq.vq_lock, flags);
 
